components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -1263,22 +1259,17 @@ public boolean isUsernameAlreadyTaken(String username) throws IdentityRecoveryEx
     public boolean isUsernameAlreadyTaken(String username, String tenantDomain) throws IdentityRecoveryException {
 
         boolean isUsernameAlreadyTaken = true;
-        WorkflowManagementService workflowService = new WorkflowManagementServiceImpl();
         if (StringUtils.isBlank(tenantDomain)) {
             tenantDomain = MultitenantUtils.getTenantDomain(username);
         }
         try {
             String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
-            Entity userEntity = new Entity(tenantAwareUsername, IdentityRecoveryConstants.ENTITY_TYPE_USER,
-                    IdentityTenantUtil.getTenantId(tenantDomain));
 
             UserRealm userRealm = getUserRealm(tenantDomain);
             if (userRealm != null) {
-                isUsernameAlreadyTaken = userRealm.getUserStoreManager().isExistingUser(tenantAwareUsername) ||
-                        workflowService.entityHasPendingWorkflowsOfType(userEntity,
-                                IdentityRecoveryConstants.ADD_USER_EVENT);
+                isUsernameAlreadyTaken = userRealm.getUserStoreManager().isExistingUser(tenantAwareUsername);
```

Comment:
How the system will know if there is a user creation approval is pending for this username? do we think we dont need such requirement?

Reply:
We are removing workflow feature from IPK, this will not affect Identity Server